### ISSUE TITLE
Error on editing large number of items

### ISSUE THREAD

**Issue by sid-kap** [burke-software/django-mass-edit#33]
It throws a 500 error when I click save after I mass-edit 100 items or more.
I don't see anything in the logs.

**Comment by PetrDlouhy** [comment]
For anyone bumping here, I had similar problem with Nginx. It is caused by too long urls generated by mass-edit, when trying to change many (~3000) records. I had following record in Nginx error log:
```upstream sent too big header while reading response header from upstream```

The solution was setting Nginx to accept bigger URL:

```
large_client_header_buffers 4 32k;
proxy_buffer_size          32k;
```

and `--limit-request-line` setting in Gunicorn.

Although this solution worked in my case, I don't consider it a complete solution. I think, that mass-edit should change it's behaviour so that very long URLs are avoided, because it is slow and hacky.

**Comment by gamesbook** [comment]
Do you think a similar solution is possible with Apache?

**Comment by PetrDlouhy** [comment]
gamesbook: It is more workaround than solution, because the URL size is still limited (only bigger). I am thinking about rewriting mass-edit in such way, that it stores object IDs in session or hidden field or something.

**Comment by PetrDlouhy** [comment]
Rewrite mass-edit to store object ids in session was very easy. It is in my use_sessions branch now: https://github.com/PetrDlouhy/django-mass-edit/tree/use_sessions

@gamesbook: Could you please test it before I make pull request from this?

**Comment by variant77** [comment]
The issue is also fixed in #62 it is also available in https://github.com/variant77/django-mass-edit
